framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -480,6 +480,14 @@ protected String standardizeString(
             if (result == null) {
                 return new DependentTypesError(expression, " ").toString();
             }
+            if (result instanceof FlowExpressions.FieldAccess
+                    && ((FlowExpressions.FieldAccess) result).isFinal()) {
+                Object constant =
+                        ((FlowExpressions.FieldAccess) result).getField().getConstantValue();
```

Comment:
Doubles and floats aren't parseable, so if the constant value is a double or float, the name of the field should be used instead.